### ISSUE TITLE
Muzzle not working with screen recording apps in Catalina

### ISSUE THREAD

**Issue by rafiron** [bdkjones/MuzzleApp#69]
When sharing my screen using hangouts via chrome (permissions granted to Google Chrome under 'Screen Recording' section in Security & Privacy settings), the app doesn't turn on the DnD. I tried with some other apps but the behaviour is the same.
I've also checked that the application has been granted 'Accessibility' permissions.

Please let me know if I'm missing out on some setting or something else is needed to make the application work properly.

Here are some additional details :
Chrome version : 80.0.3987.116 
MacOS version : Catalina 10.15.3
Muzzle app version : Version 1.5 Build 304

**Comment by bdkjones** [comment]
A few people have reported this with Hangouts now. I'm looking into it!

**Comment by dct65** [comment]
I've seen this same issue recently on Catalina with BlueJeans.  Interesting additional bit of info is that with AdobeConnect, Muzzle still works fine on the same machine.

BlueJeans: Version 2.18.0 (2.18.0.272)
MacOS: 10.15.3 (19D76)
Muzzle: Version 1.5 Build 304

**Comment by rafiron** [comment]
@bdkjones , in one of the issues someone mentioned that muzzle should also have System events access enabled under Security & Privacy -> Automation. However, I can't see Muzzle in that section. 

I tried re-installing the app but it seem to only ask for Accessibility permission and not for access to System events. Can that be an issue causing the app to not function properly ?

**Comment by bdkjones** [comment]
Yes. That would be an issue. Catalina is basically a giant dumpster fire of security-theater. I hope Apple course-corrects for 10.16 because the current approach is awful.

You can use this command to reset Muzzle’s security permission prompt. MacOS will allow the prompt only one time, ever. After that, there’s no way for the app to ever show you the prompt again without a hard reset using this command:

    tccutil reset All com.incident57.Muzzle

Run that with Muzzle NOT running (be sure it’s really not running; not just hidden in the UI). Reboot your Mac afterwards. When you launch Muzzle, it should prompt you for security access and you’ll then have to approve it in the list.

**Comment by rafiron** [comment]
@bdkjones  I tried the same thing previously as well (did it again just for a good measure), but when the app starts up, it only asks for Accessibility permission. I still couldn't see the app requesting System events' access.
Can you check if it's the app is working properly in Catalina and if there is some workaround that can be added to ensure it gets the required events' data.

**Comment by rafiron** [comment]
@bdkjones do you have any other suggestions that I can try out. Or do you feel it's an issue with Catalina's under the hood security changes.

**Comment by bdkjones** [comment]
I have a feeling it's a Catalina thing, but I need to get some time to dig into this and look at it in depth.

**Comment by geojohn23** [comment]
Hi Bryan,

Any update on fixing this? I'm also on Catalina (10.15.6 (19G73)) and Muzzle appears to never stop notifications when using Zoom, GTM, or Meet.  I see Muzzle in the Security&Privacy->Privacy->Accessibility list and it's checked. For good measure, I killed Muzzle did the tccutil reset, cleared Muzzle out of the Accessibility list and started over and still Muzzle is not stopping notifications.  Would really love to get this working (for all those same reasons as to why it was created).

**Comment by bdkjones** [comment]
If you’re not seeing Muzzle work with Zoom, you probably haven’t turned off the Zoom option that auto-toggles DND when screensharing starts. That setting conflicts with Muzzle and ends up toggling DND twice.

-Bryan

**Comment by geojohn23** [comment]
THAT WAS IT!  Thanks!

You are my hero.


------------------
John Will
http://www.linkedin.com/in/johnwill

**Comment by paul-brenner** [comment]
I don't think I understand, has this been fixed for all apps? I use Brave and google hangouts. Muzzle hasn't worked for me in Catalina. I also see Muzzle checked in Security&Privacy->Privacy->Accessibility. I've been following this issue with fingers crossed!

**Comment by geojohn23** [comment]
It works for me now for Zoom and Google Meet (I could have sworn it wasn't working for Meet before, but it is now) -- here's what I did.

Open Terminal window and execute:

> $ killall Muzzle
> $ sudo tccutil reset All com.incident57.Muzzle
> $ open -b com.apple.systempreferences /System/Library/PreferencePanes/Security.prefPane/

Then, in the security pane that comes up in the command above, Click to Unlock security settings, head to Accessibility and click the "-" to remove Muzzle. Launch Zoom, click settings, choose "Screen Share" settings and uncheck the option "Silence system notifications when sharing desktop". 
![image](https://user-images.githubusercontent.com/68920817/88721382-3abb7380-d0db-11ea-8d60-cf4313c39b42.png)
 
Then re-launch Muzzle and when it launches, approve security access.

Now - check the system's Do Not Disturb setting (click on the notification icon in the MAC toolbar and scroll up to see the DND setting) and see that it is off.  
![image](https://user-images.githubusercontent.com/68920817/88721572-80783c00-d0db-11ea-8dcd-12384f8d1fd1.png)

Now Start a Zoom session and start sharing your screen - check again now the system Do Not Disturb setting and see that it is now on.  
![image](https://user-images.githubusercontent.com/68920817/88722044-b0bfda80-d0db-11ea-9183-80baab3db754.png)
Exit Zoom.  Check the system DND setting and it is back to off.  

Now start a Google Meet and share your screen and again check the system DND setting and is now on.

So, just starting a Zoom or Meet doesn't turn on the DND -- it is only when you screen share from those that DND gets turned on.

Hope this helps

**Comment by bdkjones** [comment]
I haven’t tried Brave, so I can’t confirm that browser. 

-Bryan

**Comment by bdkjones** [comment]
That’s correct. I can’t count the number of times folks have misunderstood that Muzzle kicks in ONLY when you start sharing your screen. 

-Bryan

**Comment by paul-brenner** [comment]
I am so guilty right now. My head is hanging in the most shame. I'm that person who forgot that Muzzle only kicks in during screen sharing. Looks like the steps above to get muzzle into Security&Privacy->Privacy->Accessibility did fix things and I can confirm it works. 

Not saying that you would want to do this, but would it even be possible to make Muzzle turn on DND during video calls or is it only reasonably possible when screen sharing starts? 

Again, a million apologies for being one of those people who didn't even realize muzzle is completely working correctly.
